### DISCUSSION TITLE
Regarding mnist example case

### DISCUSSION THREAD

**Discussion by ravikiran19980** [modern-fortran/neural-fortran#91] [Q&A]
Hello I went through the codes and I had a small doubt with respect to the mnist example case. The input data is of size 784 . That should be after flattening the 28x28 image right? But I do not see the flattening used in that example. Or iam I missing something? It would be great if you could kindly let me know. 
Thank you

**Comment by milancurcic** [comment]
Hi @ravikiran19980, sorry for the late response. I read this at a weird time and later forgot to respond.

Yes, in a dense network where the input layer is 1-d, we input 1-d data. In the MNIST case, the input data is already 1-d (size 784), even though they represent 28x28 pixel images. That's why there's no flattening in the example--the data is already flat. Dense networks don't have a concept of spatial patterns and each pixel is treated independently from others, unlike in convolutional networks.

Does this answer your question?
